feature/system/gnmi/cliorigin/mixed_oc_cli_origin_support_test/mixed_oc_cli_origin_support_test.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package mixed_oc_cli_origin_support_test implements GNMI 1.12 from go/wbb:vendor-testplan
+package mixed_oc_cli_origin_support_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ygot/ygot"
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+func interfaceDescriptionCLI(dp *ondatra.Port, desc string) string {
+	switch dp.Device().Vendor() {
+	case ondatra.ARISTA, ondatra.CISCO:
+		const tmpl = `
+interface %s
+  description %s
+`
+		return fmt.Sprintf(tmpl, dp.Name(), desc)
+	}
+	return ""
+}
+
+func buildOCUpdate(path *gpb.Path, value string) *gpb.Update {
+	if len(path.GetElem()) == 0 || path.GetElem()[0].GetName() != "meta" {
+		path.Origin = "openconfig"
+	}
+	update := &gpb.Update{
+		Path: path,
+		Val:  &gpb.TypedValue{Value: &gpb.TypedValue_StringVal{StringVal: value}},
+	}
+	return update
+}
+
+func buildCLIUpdate(value string) *gpb.Update {
+	update := &gpb.Update{
+		Path: &gpb.Path{
+			Origin: "cli",
+			Elem:   []*gpb.PathElem{},
+		},
+		Val: &gpb.TypedValue{
+			Value: &gpb.TypedValue_AsciiVal{
+				AsciiVal: value,
+			},
+		},
+	}
+	return update
+}
+
+// Push overlapping mixed SetRequest specifying CLI before OpenConfig for  DUT port-1.
```

Comment:
The comment says CLI before OpenConfig but the request below puts OpenConfig before CLI. In my suggested edit I'm going with this comment.

Reply:
Ack. Done, committed all changes